### DISCUSSION TITLE
7-Segment Display TM1637 via ESP01 and MQTT on Home Assistant for newbees

### DISCUSSION THREAD

**Discussion by karefor** [arendst/Tasmota#21231] [Show and tell]
To display the values via MQTT and 7-Segment Display I used gitpod and tasmota master to compile and added the lines

#ifndef USE_DISPLAY
#define USE_DISPLAY
#endif

#ifndef USE_DISPLAY_TM1637
#define USE_DISPLAY_TM1637
#endif
(from https://tasmota.github.io/docs/TM163x/)    click ">"

and if you need "MQTT Subscribe" also add: 
#ifndef SUPPORT_MQTT_EVENT
#define SUPPORT_MQTT_EVENT
#endif
(from https://tasmota.github.io/docs/MQTT/#clearing-retained-messages)

both in user_config_override.h before the last "endif".

There is a nice video on youtube from vccground where he explains how to compile: (from minute 2:20): 
https://www.youtube.com/watch?v=WashxTcHiDc  
***************************************************************************************************************************************
To get shown something on the Display write in Tasmota console e.g.: 
DisplayNumber 1234    
(from https://tasmota.github.io/docs/TM163x/#tm1638-leds-and-buttons)
***************************************************************************************************************************************
How to send the values from Home Assistant to your display-ESP: 
Via Einstellungen/Automatisierungen & Szenen Automatisierung erstellen   neue Automatisierung erstellen
choose "when entity changes state or any attributes"
Dann mache / Aktion hinzufügen / (here search for "mqtt") choose "MQTT: Veröffentlichen"

at Topic I used:

cmnd/tasmota_200D02/Displaynumber

(tasmota_200D02 has to be replaced with the topic name of y o u r ESP. You will find it at Configuration/Configure MQTT. It is the name after %topic% (tasmota_200D02) in the brackets.)

at Nutzlast-Vorlage I used:

{{ float(states('sensor.tasmota_lk13be_current')) | round(0) }}, 0, 0, 4

(, 0, 0, 4 makes the ciphers right-aligned)
Instead of "sensor.tasmota_lk13be_current" you have to use your own sensors name.
(from https://github.com/arendst/Tasmota/discussions/19596)

In the beginning I also had some difficulties with the soldering of the connection-pins of the TM1637 because there was not enough tin solder at the data and clock pins...

Enjoy!

**Comment by karefor** [comment]
The old display stopped working. 
So I had to prepare a new one. 

This only worked with GPIO 0 for DIO.

So I used this settings on the ESP01:
Configure Module:
Generic(18) 
GPIO 0         set to TM1637 DIO
GPIO 1 (TX) set to TM1637 CLK 
all the others set to "none"

At Configure Template: 
Generic (18) 
and all set to "user"

Good Luck :-)

**Reply by 7segmentgpod** [reply]
Another issue: After some weeks the display didn't show anything. It worked again after just switching "off" to "on" via the "toggle"-button on the Main Menu....

**Reply by sfromis** [reply]
If it worked what powering the display on, that is no "issue", just something you need to ensure.

**Reply by 7segmentgpod** [reply]
Well it worked from the beginning. 
So I wonder what turned it off. 
But You're right ensuring that will be always "on" will solve the problem. Thank You!
